bin/ch/Helpers.cpp

```diff
@@ -133,9 +58,9 @@ HRESULT Helpers::LoadScriptFromFile(LPCSTR filename, LPCWSTR& contents, bool* is
     //
     // Read the entire content as a binary block.
     //
-    fread((void*)contentsRaw, sizeof(char), lengthBytes, file);
+    fread(pRawBytes, sizeof(BYTE), lengthBytes, file);
     fclose(file);
-    *(WCHAR*)((byte*)contentsRaw + lengthBytes) = _u('\0'); // Null terminate it. Could be LPCWSTR.
+    *reinterpret_cast<WCHAR*>(pRawBytes + lengthBytes) = 0; // Null terminate it. Could be UTF16
```

Comment:
> I haven't tried yet but most probably compiler already knows pRawBytes is  sizeof(BYTE)!! x (lengthBytes) away from it's starting point. So it does the rest hence it wouldn't give trouble during the runtime. However comparing to old code, it is miss-leading. Adding (BYTE*) right before pRawBytes would make it more clear.

With the new code `pRawBytes` is of type `BYTE*`, thus it does not need cast. Casting to its own type `(BYTE*)pRawBytes` is unnecessary and redundant. I'm not sure why you describe it as uncertain ("haven't tried...probably"), `pRawBytes + lengthBytes` means &pRawBytes[lengthBytes]` and yes `lengthBytes * sizeof(BYTE)`, or `lengthBytes` bytes away from start.

The old code needs casting because `contentsRaw` was declared as `LPCWSTR contentsRaw`.

> So it is okay to put null termination in a single line but IMO making it clear would give less trouble to reader.

What do you have in mind to make it clear? I agree with you this line is a bit tricky without understanding it is trying to null-terminate both Utf8 and Utf16 cases. I actually worried that if `lengthBytes` is odd and this WORD assigns to odd address and the instruction may fail, but tried out on Linux and it worked. (For context, this is test binary and the code isn't in product quality. As long as it works for our unit tests.)

> there is no guarantee that it will be 2 bytes.

You mean 'char16_t` isn't guaranteed to be 2-bytes? I can't believe... What's the use of `char16_t` if it doesn't guarantee to be `16 bits"?